pallets/identity/src/keys.rs

```diff
@@ -884,9 +884,22 @@ impl<T: Config> Module<T> {
         ensure_custom_length_ok::<T>(perms.complexity(), MAX_PERMISSION_COMPLEXITY)?;
         ensure_custom_length_ok::<T>(perms.asset.complexity(), MAX_ASSETS)?;
         ensure_custom_length_ok::<T>(perms.portfolio.complexity(), MAX_PORTFOLIOS)?;
+        Self::ensure_no_except_perms(&perms.extrinsic)?;
         Self::ensure_extrinsic_perms_length_limited(&perms.extrinsic)
     }
 
+    // Ensures that extrinsic permissions do not use the Except variant
+    // This is considered unsafe since extrinsic names can change or be replaced with newer versions
+    pub fn ensure_no_except_perms(perms: &ExtrinsicPermissions) -> DispatchResult {
+        let is_not_except = match perms {
+            SubsetRestriction::Whole => true,
+            SubsetRestriction::These(_) => true,
+            SubsetRestriction::Except(_) => false,
+        };
+        ensure!(is_not_except, Error::<T>::ExceptNotAllowedForExtrinsics);
+        Ok(())
+    }
```

Comment:
This only checks if `Except` is used at the pallet level (blacklisting some pallets).  The `ExtrinsicPermissions` has two-levels of `SubsetRestriction`.

The `dispatchable_names` field is also a `SubsetRestriction`, where we also need to check for `Except`.

Reply:
`ExtrinsicPermissions` is a bad type name, since it is really the Pallet permissions.  Then we have `DispatchableNames` type for the Extrinsic permissions.